dashboard/app/models/script.rb

```diff
@@ -1845,17 +1845,26 @@ def get_course_version
     course_version || unit_group&.course_version
   end
 
-  # If a script is in a unit group, use that unit group's published state. If not, use the script's published_state
-  # If both are null, the script is in_development
-  def get_published_state
-    published_state || unit_group&.published_state || SharedConstants::PUBLISHED_STATE.in_development
-  end
-
   # Use the unit group's pilot_experiment if one exists
   def get_pilot_experiment
     pilot_experiment || unit_group&.pilot_experiment
   end
 
+  # Get the published state on the course version and default to in_development if it doesn't exist
+  def get_published_state
+    return published_state if published_state
+    return unit_group.get_published_state if unit_group
+    return course_version&.published_state if course_version
+    SharedConstants::PUBLISHED_STATE.in_development
```

Comment:
I think I am a bit lost on why this is the right logic, in part because it's unclear to me what cases we care about and which we don't. would you be willing to add some unit tests on this method to help clarify?